gke-platform/modules/kubernetes/kubernetes.tf

```diff
@@ -16,7 +16,7 @@ data "http" "nvidia_driver_installer_manifest" {
   url = "https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml"
 }
 
-resource "kubectl_manifest" "nvidia_driver_installer" {
-  yaml_body = data.http.nvidia_driver_installer_manifest.response_body
-  count = var.enable_tpu || var.enable_autopilot ? 0 : 1
+resource "kubernetes_manifest" "nvidia_driver_installer" {
```

Comment:
How does this work? My understanding was that `kubernetes_manifest` wouldn't work if the cluster was not yet created, but the cluster is also being created as part of the gke-platform module, right?

Reply:
I see, it looks like this is now being called by the ray user template, rather than the `gke-platform/main.tf`? In that case, I think this module (and maybe also the kuberay one?) should be moved out of the gke-platform directory.

Reply:
So my understanding is when we say platform, its mostly the base setup required in terms of GCP services, something like, provision, GKE, nodepools, IAM, IAP, etc..

and the workloads (or applications) is something which will be deployed on kubernetes engine as container/services/ingress/etc. 

So, ideal pipeline would be like:
1. provision platform:  provision GKE with nodepools, setup required IAM, enable APIs, etc.
2. Provision workloads: now GKE is ready, deploy all workloads like ray-operator, ray-cluster, jupyter-hub, stable-difussion, etc.. using kubernetes providers.

In that way the `kubernetes_manifest` will also work as GKE got prepared in first step.